src/tqec/circuit/detectors/utils.py

```diff
@@ -0,0 +1,571 @@
+
+    Returns:
+        `True` if the provided moment has a reset, else `False`.
+    """
+    for inst in moment:
+        if _is_virtual_instruction(inst):  # type: ignore
+            continue
+        if not _is_reset(inst):  # type: ignore
+            return False
+    return True
+
+
+def is_virtual_moment(moment: stim.Circuit) -> bool:
+    """Check if the provided moment only contains virtual instructions.
+
+    Virtual instructions are noisy instructions or instructions whose name is
+    in the `ANNOTATIONS` structure.
+
+    Args:
+        moment: circuit to check.
+
+    Returns:
+        `True` if the provided circuit is only composed of virtual instructions,
+        else `False`.
+    """
+    if any(isinstance(inst, stim.CircuitRepeatBlock) for inst in moment):
+        raise TQECException(
+            "Breaking invariant: you provided a circuit with a stim.CircuitRepeatBlock "
+            "instruction to is_virtual_moment. This is not supported."
+        )
+    return all(_is_virtual_instruction(inst) for inst in moment)  # type:ignore
+
+
+def split_combined_measurement_reset_in_moment(
+    moment: stim.Circuit,
+) -> tuple[stim.Circuit, stim.Circuit]:
+    """Split a moment that contains combined operations into two moments.
+
+    The moment should only contain annotations, noisy operations, or combined
+    measurement/reset operations.
+
+    Warning:
+        This function assumes that any annotation encountered in the provided
+        moment should appear just after all the measurements:
+        ```
+        measurements - TICK - resets
+                     ^
+            All annotations are inserted here.
+        ```
+        This assumption seems reasonable for `DETECTOR` and `OBSERVABLE_INCLUDE`
+        annotations, which are expected to be the most commonly found in a
+        combined measurement/reset moment.
+        Nevertheless, be aware of that limitation when using that function.
+
+    Args:
+        moment: a valid moment (i.e., sub-circuit contained between two TICK
+            instructions) containing only measurement or combined instructions.
+
+    Raises:
+        TQECException: if the provided moment contains a stim.CircuitRepeatBlock
+            instruction.
+
+    Returns:
+        two `stim.Circuit` instances `(measurements, resets)` that repectively
+        contain all measurements and all resets found in the provided `moment`.
+        The returned `measurements` circuit is guaranteed to finish by a TICK
+        instruction.
+        The returned `resets` circuit is guaranteed to not finish by a TICK
+        instruction.
+        `measurements` is guaranteed to contain all the instructions that appeared
+        before the combined operation.
+        `resets` is guaranteed to contain all the instructions that appeared
+        after the combined operation.
+
+    """
+    measurements = stim.Circuit()
+    resets = stim.Circuit()
+
+    measured_qubits: set[int] = set()
```

Comment:
I do not understand why `set[int]` is not fine. Maybe because `from __future__ import annotations` was not included before, it is now and the `set[int]` works. 
I will let you resolve that conversation, just in case you had another reason to change that annotation.

Reply:
I will resolve all the other conversations about such changes, keeping that one open if you had another reason to request for such a change.

Reply:
It's all fine, I started reviewing before you fixed this and honestly forgot that you could do the `__future__` import